src/lib/support/logging/CHIPLogging.cpp

```diff
@@ -218,12 +218,13 @@ void LogV(uint8_t module, uint8_t category, const char * msg, va_list args)
 
     if (redirect != nullptr)
     {
-        redirect(moduleName, category, msg, args);
-    }
-    else
-    {
-        Platform::LogV(moduleName, category, msg, args);
+        // if the redirect is going to read these args we need to make a copy
+        // of them otherwise the va_list may be modified and unreadable
+        va_list copy;
```

Comment:
i'm not in love with this, would appreciate suggestions on how we might handle this better if we need to make the call to both the platform logger as well as the redirected logger